uint16 rate
---
bool success
string message